Automattic/woocommerce-payments
# Store balance transaction ID in order metadata

Fixes #7856

#### Changes proposed in this Pull Request

* Add a new key `_wcpay_payment_transaction_id` in order metadata to store Stripe's balance transaction ID for each payment to aid in transaction reconciliation in third-party accounting systems.
* Refactor the `attach_intent_info_to_order` method to make it easier to add new intent-based data values to order meta.

#### Testing instructions

* Place a new order in WooCommerce frontend with any payment method.
* Make a GET request to `/wp-json/wc/v3/orders/<order_id>` (see [Retrieve an order API](https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order)).
* Verify that there is a new key `_payment_transaction_id` in the response order object's meta_data property with a value starting with 'txn_'.
* Also verify in WP Admin that the transaction id for the above order on the Transactions page matches the value in the previous step.


https://github.com/Automattic/woocommerce-payments/assets/1288616/c5bbc618-0588-40dd-998d-df38314aea8f

-------------------


**Post merge**
